File: jvm/mongo-hostname-verification-disabled.java
package test

import com.mongodb.MongoClientSettings
import com.mongodb.MongoClients

class HelloWorld {
    public static void main(String[] args) {
        MongoClientSettings settings = MongoClientSettings.builder()
            .applyToSslSettings(builder -> {
                builder.enabled(true);
                // ruleid: mongo-hostname-verification-disabled
                builder.invalidHostNameAllowed(true);
            })
            .build();

        MongoClientSettings settings = MongoClientSettings.builder()
            .applyToSslSettings(builder -> {
                // ok: mongo-hostname-verification-disabled
                builder.enabled(true);
            })
            .build();
    }
}

File: kotlin/mongo-hostname-verification-disabled.kt
package test

import com.mongodb.MongoClientSettings
import com.mongodb.MongoClients

fun main() {
    val mongoClient = MongoClient.create(
        MongoClientSettings.builder()
            .applyConnectionString(ConnectionString("<your connection string>"))
            .applyToSslSettings{ builder ->
                builder.enabled(true)
                // ruleid: mongo-hostname-verification-disabled
                builder.invalidHostNameAllowed(true);
            }
            .build()
    )

    val mongoClient = MongoClient.create(
        MongoClientSettings.builder()
            .applyConnectionString(ConnectionString("<your connection string>"))
            .applyToSslSettings{ builder ->
                // ok: mongo-hostname-verification-disabled
                builder.enabled(true)
            }
            .build()
    )
}
